packages/backend-data/src/convert_schema.ts

```diff
@@ -40,24 +94,124 @@ export const convertSchemaToCDK = (
       functionSlots,
       dataSourceStrategies: generatedModelDataSourceStrategies,
+      customSqlDataSourceStrategies:
+        customSqlDataSourceStrategies?.map(
+          (existing: CustomSqlDataSourceStrategy) => ({
+            ...existing,
+            strategy: dbStrategy,
+          })
+        ) || [],
+    };
+  }
+
+  return AmplifyDataDefinition.fromString(schema, DYNAMO_DATA_SOURCE_STRATEGY);
+};
+
+/**
+ * Given an input list of CDK Graphql Def interface schemas, produce the relevant CDK Graphql Def interface
+ * @param schemas the cdk graphql definition interfaces to combine
+ * @returns the cdk graphql definition interface
+ */
+export const combineCDKSchemas = (
+  schemas: IAmplifyDataDefinition[]
+): IAmplifyDataDefinition => {
+  return AmplifyDataDefinition.combine(schemas);
+};
+
+/**
+ * Given the generated rds builder database configuration, convert it into the DataSource strategy
+ * @param configuration the database configuration from `data-schema`
+ * @returns the data strategy needed to configure the data source
+ */
+const convertDatabaseConfigurationToDataSourceStrategy = (
+  configuration: DataSourceConfiguration,
+  customSqlDataSourceStrategies: CustomSqlDataSourceStrategy[] = [],
+  backendSecretResolver: BackendSecretResolver,
+  provisionStrategyName: string
+): ModelDataSourceStrategy => {
+  if (configuration.engine === 'dynamodb') {
+    return DYNAMO_DATA_SOURCE_STRATEGY;
+  }
+
+  const dbType = RDS_DB_TYPES[configuration.engine];
+  let vpcConfiguration: VpcConfig | undefined = undefined;
+
+  if (configuration.vpcConfig !== undefined) {
+    vpcConfiguration = {
+      vpcId: configuration.vpcConfig.vpcId,
+      securityGroupIds: configuration.vpcConfig.securityGroupIds,
+      subnetAvailabilityZoneConfig:
+        configuration.vpcConfig.subnetAvailabilityZones,
     };
   }
 
-  return AmplifyDataDefinition.fromString(schema, {
+  const customSqlStatements = customSqlStatementsFromStrategies(
+    customSqlDataSourceStrategies
+  );
+
+  return {
     dbType,
-    provisionStrategy,
-  });
+    name: provisionStrategyName + configuration.engine,
+    dbConnectionConfig: {
+      connectionUriSsmPath: backendSecretResolver.resolvePath(
+        configuration.connectionUri
+      ).branchSecretPath,
+    },
+    vpcConfiguration,
+    customSqlStatements,
+  };
+};
+
+/**
+ * Create a custom sql statement reference dictionary
+ * @param customSqlDataSourceStrategies custom sql handler defined in the schema
+ * @returns an object mapping the file path to the sql statement
+ */
+const customSqlStatementsFromStrategies = (
+  customSqlDataSourceStrategies: CustomSqlDataSourceStrategy[]
+): Record<string, string> => {
+  const customSqlStatements = customSqlDataSourceStrategies
+    .filter((sqlStrategy) => sqlStrategy.entry !== undefined)
+    .reduce((acc, sqlStrategy) => {
+      const entry = sqlStrategy.entry as JsResolverEntry; // undefined is filtered out above
```

Comment:
Could we use `sqlStrategy.entry!` instead? That has less of a chance of hiding type errors as the code evolves